File: cinemaProject/src/main/java/com/example/cinemaProject/model/Bilet.java
package com.example.cinemaProject.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
public class Bilet {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch=FetchType.EAGER)
    private Movie filmDinBilet;

    private int pret=0;
    private String locInSala="--";

    @ManyToOne(fetch=FetchType.EAGER,cascade = CascadeType.REMOVE)
    @JoinColumn(name="fk_bilet_to_cosCumparaturi")
    private CosCumparaturi cosCumparaturi;
}

File: src/app/model/Bilet.ts
import {Movie} from "./Movie";

export class Bilet{
  id: number | undefined;
  locInSala:string | undefined;
  pret:number | undefined;
  movie:Movie|undefined;

  constructor(id: number | undefined, locInSala: string | undefined, pret: number | undefined, movie: Movie | undefined) {
    this.id = id;
    this.locInSala = locInSala;
    this.pret = pret;
    this.movie = movie;
  }
}
